### DISCUSSION TITLE
feature request: query string should obey route definition

### DISCUSSION THREAD

**Discussion by thyseus** [livewire/livewire#4454] [1. Help]
Hey,

i have a quite straightforward feature idea:

given the route
```php
Route::get('/{search}', function (string $search) { ... });
```

and the query string definition 
```php
    public string $search = '';
    protected $queryString = ['search' => ['except' => '']];
```

i want the query string url to be generated by the livewire component to look like

http://localhost/search-term

instead of

http://localhost/?search=search-term

which it is doing right now. Is there a possibility to override the query string generation by the developer manually _OR_ would it be possible to let livewire automagically see if there are route parameters in the defined route and fill in these route parameters instead of appending a `?foo=bar` query string?

thanks a lot !

**Comment by joshhanley** [comment]
@thyseus Livewire already supports route parameters, but it's different to the queryString. So try removing 'search' from `$queryString` and you should find that if you browse to `/search-term` that `$search` should already be set to that.

Hope this helps!
